python/pyarrow/fs.py

```diff
@@ -227,16 +227,34 @@ def copy_files(source, destination,
 
     Examples
     --------
-    Copy an S3 bucket's files to a local directory:
+    Inspect an S3 bucket's files:
 
-    >>> copy_files("s3://your-bucket-name",
-    ...            "local-directory") # doctest: +SKIP
+    >>> s3, path = fs.FileSystem.from_uri(
+    ...            "s3://registry.opendata.aws/roda/ndjson/")
+    >>> selector = fs.FileSelector("registry.opendata.aws/roda/ndjson")
+    >>> s3.get_file_info(selector)
+    [<FileInfo for 'registry.opendata.aws/roda/ndjson/index.ndjson':...]
 
-    Using a FileSystem object:
+    Create a LocalFIleSystem object and a new directory:
 
-    >>> copy_files("your-bucket-name", "local-directory",
-    ...            source_filesystem=S3FileSystem(...)) # doctest: +SKIP
+    >>> local = fs.LocalFileSystem()
+    >>> local.create_dir('/tmp/copy-dir')
```

Comment:
Is all this boilerplate really useful? Perhaps it's not important to make the examples executable, as long as they are informative? Here we want to showcase usage of `copy_files`, not the rest of the filesystem API.

Reply:
My position would be: if it's difficult to make an example both readable and executable, then readability should take the upper hand. YMMV, though :-)